Snyk test all okay-replaced start command-updated readme

juliaguida opened the pull request:
## Changes

Updated the readme with instruction to run the app locally.

I have test this app with the npm run test command which triggers snyk. The dependency is no longer displayed under the list of Vulnerabilities requiring update.

Why were these changes necessary? What problem are you trying to solve? Link to JIRA ticket ?
https://financialtimes.atlassian.net/browse/CRM-3404

## Test it locally:
copy this branch 
Replace the start command with "start": "doppler run -p ft-tps-screener -c prod -- node app.js" in package.json.
**_Note: Ensure the start command is reverted to its original state before merge into prod._**
Run `doppler login` command
Run `npm run postinstall ` command
Run `npm run start` command
Enter a UK number in the browser's search bar; if it's registered, it's important to refrain from contacting for sales and marketing purposes.
After changes localhost:3000 working as expected
![Screenshot 2024-06-11 at 12 14 02](https://github.com/Financial-Times/tps-lookup/assets/32463353/a6077800-cbac-4021-9ecf-dd45ee59e9ce)
Snyk test result:
![Screenshot 2024-06-11 at 12 07 42](https://github.com/Financial-Times/tps-lookup/assets/32463353/1a60acb0-0949-4419-9542-44a79b91b2ad)

melanierogan commented:
Hi Julia, you’ve removed the updated the npm start command, I understand why, but I think you can update the readme to use the start:dev command to run the app locally.

Joemusarurwa commented on README.md:
````diff
@@ -12,32 +12,23 @@ Then, an FT service can internally query the TPS Screener to filter the list of
-npm start
-```
+Replace the start command with "start": "doppler run -p ft-tps-screener -c prod -- node app.js" in package.json.
+**_Note: Ensure the start command is reverted to its original state before merge into prod._**
````
Nice work. Suggest an additional script (feel free to ignore if this is not preferred) - perhaps something like `"start:prod-from-local": "doppler run -p ft-tps-screener -c prod -- node app.js"` so that one does not have to change the start script every time.

Joemusarurwa commented on README.md:
```diff
@@ -12,32 +12,23 @@ Then, an FT service can internally query the TPS Screener to filter the list of
+Run `doppler login` command
+Run `npm run postinstall ` command
+Run `npm run start` command
+Enter a UK number in the browser's search bar; if it's registered, it's important to refrain from contacting for sales and marketing purposes.
```
```suggestion
Go to `http://localhost:3000/` in your browser and enter a UK number in the browser's search bar; if it's registered, it's important to refrain from contacting for sales and marketing purposes.
```

juliaguida replied:
Thank you ✅

awar7118 commented on package.json:
```diff
@@ -10,7 +10,7 @@
     "start": "node app.js",
     "prepare": "npm run snyk-protect; npm run snyk-protect",
```
Not something you introduced, but there are a few unnecessary repetitions of snyk commands that can be consolidated/improved(you don't need to do it in this PR, but something worth mentioning)

```
"test": "snyk test",
"prepare": "npm run snyk-protect; npm run snyk-protect",
"snyk-protect": "snyk-protect",
```

juliaguida replied:
Thanks Ahmed for your suggestion I implemented it and will be pushed on my next commit.

juliaguida commented on .snyk:
Thanks @awar7118 I will rephrase it. When I run the snyk test or npm run test in the terminal after ignoring the "No fix" remediations path available, I no longer see prompts to update vulnerabilities that could potentially break the tests in CircleCI. Please see the screenshot below:
Snyk test result:
![Screenshot 2024-06-11 at 12 07 42](https://github.com/Financial-Times/tps-lookup/assets/32463353/1a60acb0-0949-4419-9542-44a79b91b2ad)

juliaguida commented:
> Hi Julia, you’ve removed the updated the npm start command, I understand why, but I think you can update the readme to use the start:dev command to run the app locally.

Thanks @melanierogan for your feedback, the original readme points to npm start as an accurate way of testing. 
Which remains useful. I haven't added it to the package.json. 
However, I have added instructions pointed to it on the updated readme. For ensuring that tests are as close to production results as possible. The dev command isn't as accurate as the npm start. I introduced the dev command solely to replace the outdated vault env path.

juliaguida commented on README.md:
````diff
@@ -12,32 +12,23 @@ Then, an FT service can internally query the TPS Screener to filter the list of
-```shell
-npm run postinstall
+- Node ^18.x.x
+- [Doppler](https://github.com/Financial-Times/ip-ftlive-api#doppler---secrets-management): If new to Doppler, please follow the instruction to install the [CLI](https://docs.doppler.com/docs/install-cli).
````
Thanks Ahmed for the suggestion this is now implemented and will be pushed on the next commit.

melanierogan commented:
> > Hi Julia, you’ve removed the updated the npm start command, I understand why, but I think you can update the readme to use the start:dev command to run the app locally.
> 
> Thanks @melanierogan for your feedback, I found this out... Happy to remove the dev command for now as per the issue below. If you agree. The original readme points to the npm start as an accurate way of testing which remains useful. To ensure that tests are as close to production results as possible.
> 
> The dev command on the other hand still has the previous issue mentioned on this slack thread [tps-screener-incident-3yrs-ago] (https://financialtimes.slack.com/archives/C0254SZ9QLR/p1623412957018500) and this [thread](https://financialtimes.slack.com/archives/C0254SZ9QLR/p1623410308000200url) From running **app locally** endpoint [localhost:3000](http://localhost:3000/) error in terminal : UnrecognizedClientException: The security token included in the request is invalid. According to the thread they managed to get the app up and running but the issue still persists.

Thank you for this Julia. 
As we discussed, this will be down Okta not being set up correctly for local/dev environment. We can tackle in another ticket.

melanierogan reviewed (approved):
Nice work Julia
🚀
